openedx/core/djangoapps/content/course_overviews/management/commands/tests/test_simulate_publish.py

```diff
@@ -147,3 +154,42 @@ def sample_receiver_1(self, sender, course_key, **kwargs):  # pylint: disable=un
     def sample_receiver_2(self, sender, course_key, **kwargs):  # pylint: disable=unused-argument
         """Custom receiver for testing."""
         self.received_2.append(course_key)
+
+    def test_args_from_database(self):
+
+        # Nothing in the database, should default to disabled
+        with self.assertRaisesRegex(CommandError, 'SimulateCourseConfigPublish is disabled.*'):
+            call_command('simulate_publish', '--args-from-database')
+
+        # Add a config
+        config = SimulateCoursePublishConfig.current()
+        config.arguments = '--delay 20 --dry-run'
+        config.enabled = True
+        config.save()
+
+        with LogCapture(LOGGER_NAME) as log:
+            call_command('simulate_publish')
+
+            log.check_present(
+                (
+                    LOGGER_NAME, 'INFO',
+                    u"simulate_publish starting, dry-run={}, delay={} seconds".format('False', '0')
+                ),
+            )
+
+        with LogCapture(LOGGER_NAME) as log:
+            call_command('simulate_publish', '--args-from-database')
+
+            log.check_present(
+                (
+                    LOGGER_NAME, 'INFO',
+                    u"simulate_publish starting, dry-run={}, delay={} seconds".format('True', '20')
+                ),
+            )
+
+        config.enabled = False
+        config.save()
+
+        # Explicitly disabled
+        with self.assertRaisesRegex(CommandError, 'SimulateCourseConfigPublish is disabled.*'):
```

Comment:
We don't need to explicitly check for disabled scenario because we are already doing this above and config models return default value if object doesn't exist in DB.